doc/release/1.12.0-notes.rst

```diff
@@ -23,6 +23,20 @@ Future Changes
 Compatibility notes
 ===================
 
+DeprecationWarning to error
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Indexing with floats raises IndexError,
+  e.g., a[0, 0.0].
+* Indexing with non-integer array_like raises ``IndexError``,
+  e.g., ``a['1', '2']``
+* Indexing with multiple ellipsis raises ``IndexError``,
+  e.g., ``a[..., ...]``.
+* Indexing with boolean where integer expected raises ``IndexError``,
+  e.g., ``a[False:True:True]``.
```

Comment:
Actually, i expect this is a bad example, because it probably works again due to the code simplification of using the python slice logic. I don't mind about it, bools in slices are not really ambiguous.